### ISSUE TITLE
[fixed ONLY in demo because this is wanted behaviour] XSS vulnerability in the settings page

### ISSUE THREAD

**Issue by C0deBr8kr** [robiso/wondercms#27]
**Exploit Title:** Stored XSS vulnerability possible in settings page in WonderCMS (v2.0.3)
**Date:** 23-April-2017
**Exploit Author:** @C0deBr8kr
**Software Link:** https://www.wondercms.com/latest
**Version:** 2.0.3

**Description:**
XSS allows an attacker to run arbitrary scripts on the users browser.

**Exploit POC:**
_Browser used:_ Chrome version 57.0.2987.133

1. Login as the admin user.
2. Go to the settings page.
3. Enter <script>alert('XSS in the Page Title');</script> in  the "Page title" field.
4. Save the settings.
5. The script will get executed whenever the user navigates to the page containing the script (in this case - Home page).
6. Other fields like Page description, Main website title, Menu, Footer, Default Home page are all vulnerable to the same vulnerability of stored XSS.

**References:**
https://www.owasp.org/index.php/Cross-site_Scripting_(XSS)

**Screenshots:**

![screen shot 2017-04-23 at 1 05 57 am](https://cloud.githubusercontent.com/assets/1912754/25311956/ea5f7b40-27c1-11e7-9325-10caf2f3655a.png)

1. Entering script into page title

![screen shot 2017-04-23 at 1 14 11 am](https://cloud.githubusercontent.com/assets/1912754/25311975/3af93118-27c2-11e7-9e2b-6a5e2acb4354.png)

2. Script executes on navigating to the Home page.

**Impact:** An attacker can execute arbitrary script on an unsuspecting user's browser.

**Mitigation:** Input should be properly validated before storing in the database and output from the database should also be properly encoded before displaying it to the user.

**Comment by robiso** [comment]
@C0deBr8kr thank you for reporting this, **but this only works if the user is logged in, which we actually want for WonderCMS users. The admin should always be able to execute JavaScript in any part of WonderCMS**.

We have changed this logic in the past since we used to block out JavaScript within these fields. We don't do this any more.

If this can happen without the user being logged in, then this would be taken even more seriously.
I see this might be a problem for our demo installation of WonderCMS, which is public. **It makes sense to sanitize the input here - on the demo, like we sanitize the page parameter, since it public everywhere.**
